components/scream/src/share/io/scorpio_output.hpp

```diff
@@ -81,113 +60,124 @@
-  std::map<std::string,Int>              m_dims;
-  typename dofs_list_type::HostMirror    m_gids_host;
+  std::vector<std::string>            m_fields;
+  std::map<std::string,FieldLayout>   m_layouts;
+  std::map<std::string,int>           m_dofs;
+  std::map<std::string,int>           m_dims;
+
   // Local views of each field to be used for "averaging" output and writing to file.
+  std::map<std::string,view_1d_host>    m_host_views_1d;
+
+  // Whether this Output object writes a model restart file, or normal model output.
+  bool m_is_model_restart_output;
+
+  // If this is normal output, whether data to restart the history is needed/generated.
+  bool m_has_restart_data;
+  int m_checkpoint_freq;
 
-  std::map<std::string,view_1d_host>   m_host_views_1d;
+  // Whether this run is the restart of a previous run (in which case, we might load an output checkpoint)
+  bool m_is_restarted_run;
 
-  // Manage when files are open and closed, and what type of file I am writing.
-  bool m_is_init = false;
-  bool m_is_restart_hist = false;  //TODO:  If instead we rely on the timestamp to determine how many steps are represented in the averaging value, or maybe filename, we won't need this.
-  bool m_is_restart = false;
-  bool m_read_restart_hist = false;
+  // Whether the output file is open.
+  // Note: this is redundant, since it's equal to m_num_snapshots_in_file==0, but it makes code more readable.
+  bool m_is_output_file_open = false;
 
-  // Helper map to monitor an output stream's status.  Most used fields are Snaps and Avg Count which are
-  // used to monitor if a new file is needed and if output should be written according to the frequency settings.
-  std::map<std::string,Int> m_status = {
-                                  {"Init",          0},  // Records the number of files this output stream has managed
-                                  {"Run",           0},  // Total number of times "Run" has been called
-                                  {"Finalize",      0},  // Total number of times "Finalize" has been called (should always be 1)
-                                  {"Snaps",         0},  // Total number of timesnaps saved to the currently open file.
-                                  {"Avg Count",     0},  // Total number of timesnaps that have gone by since the last time output was written.
-                                       }; 
+  // When this equals m_out_frequency, it's time to write the output.
+  int m_nsteps_since_last_output = 0;
 
-}; // Class AtmosphereOutput
+  // When this equals m_checkpoint_freq, it's time to write a history restart file.
+  int m_nsteps_since_last_checkpoint = 0;
 
-//// ====================== IMPLEMENTATION ===================== //
-inline void AtmosphereOutput::check_status()
+  // When this equals m_out_max_steps, it's time to close the out file, and open a new one.
+  int m_num_snapshots_in_file = 0;
+};
+
+inline void AtmosphereOutput::combine (const Real& new_val, Real& curr_val) const
```

Comment:
We should add a comment to this function that explains that it controls updating variables based on the output type and is where any new output types would go.  For example if someone wanted to add standard deviation as an option, or something like that.